protoc_plugin/lib/file_generator.dart

```diff
@@ -570,12 +570,20 @@ class FileGenerator extends ProtobufContainer {
 
   /// Writes an import of a .dart file corresponding to a .proto file.
   /// (Possibly the same .proto file.)
+  ///
+  /// When generating the main file (if [forMainFile] is true), all imports
+  /// should be prefixed unless the target file is the main file. For other
+  /// files (e.g. .pbserver.dart, .pbgrpc.dart), all .pb.dart imports should
+  /// always be prefixed, even if they're for the main proto file.
   void _writeImport(IndentingWriter out, OutputConfiguration config,
-      FileGenerator target, String extension) {
+      FileGenerator target, String extension,
+      [bool forMainFile = false]) {
```

Comment:
I'd use a named optional parameter - it is easier to understand what 'true' means at the call site.

Reply:
I ended up just taking the parameter out and always prefixing as per the other suggestion.